packages/common/src/groups/HubGroup.ts

```diff
@@ -232,4 +245,69 @@ export class HubGroup
       id
     );
   }
+
+  /*
+   * Get the editor config for the HubGroup entity.
+   * @param i18nScope translation scope to be interpolated into the uiSchema
+   * @param type editor type - corresonds to the returned uiSchema
+   * @param options optional hash of dynamic uiSchema element options
+   */
+  async getEditorConfig(
+    i18nScope: string,
+    type: EditorType
+  ): Promise<IEditorConfig> {
+    // delegate to the schema subsystem
+    return getEditorConfig(i18nScope, type, this.entity, this.context);
+  }
+
+  /**
+   * Return the group as an editor object
+   */
+  toEditor(): IHubGroupEditor {
+    // cast the entity to it's editor
+    const editor = cloneObject(this.entity) as IHubGroupEditor;
+    return editor;
+  }
+
+  /**
+   * Load the group from the editor object
+   * @param editor
+   * @returns
+   */
+  async fromEditor(editor: IHubGroupEditor): Promise<IHubGroup> {
+    const isCreate = !editor.id;
+
+    // Setting the thumbnailCache will ensure that
+    // the thumbnail is updated on next save
+    if (editor._thumbnail) {
+      if (editor._thumbnail.blob) {
+        this.thumbnailCache = {
```

Comment:
This block works in the entities that extend `HubItemEntity`... but since `HubGroup` does not extend that, this will not do anything. If you look at `HubItemEntity.afterSave()` you can see how the thumbnailCache is used... so you'll need to add that logic into the `save()` method of `HubGroup`